Signal/src/view controllers/MessagesViewController.m

```diff
@@ -495,50 +492,166 @@ - (void)setNavigationTitle
     if (isGroupConversation && [navTitle length] == 0) {
         navTitle = NSLocalizedString(@"NEW_GROUP_DEFAULT_TITLE", @"");
     }
-    self.title = navTitle;
+    self.title = nil;
+
+    if ([navTitle isEqualToString:self.navigationBarTitleLabel.text]) {
+        return;
+    }
+    
+    self.navigationBarTitleLabel.text = navTitle;
+
+    // Changing the title requires relayout of the nav bar contents.
+    OWSDisappearingMessagesConfiguration *configuration =
+    [OWSDisappearingMessagesConfiguration fetchObjectWithUniqueID:self.thread.uniqueId];
+    [self setBarButtonItemsForDisappearingMessagesConfiguration:configuration];
 }
 
 - (void)setBarButtonItemsForDisappearingMessagesConfiguration:
     (OWSDisappearingMessagesConfiguration *)disappearingMessagesConfiguration
-
 {
+    UIImage *backImage = [UIImage imageNamed:@"NavBarBack"];
```

Comment:
Well, I'm going to use the custom back button everywhere. There's no safe way to consistently match (or use directly) the default back button.